src/main/java/reposense/parser/CliArguments.java

```diff
@@ -0,0 +1,130 @@
+package reposense.parser;
+
+import java.io.File;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Optional;
+
+import reposense.exception.ParseException;
+
+/**
+ * Verifies and parses user-supplied CLI arguments.
+ */
+public class CliArguments {
+
+    private static final String ARGUMENT_PREFIX_DASH = "-";
+
+    private static final String CONFIG_FILE_ARG = "config";
+    private static final String TARGET_FILE_ARG = "output";
+    private static final String SINCE_DATE_ARG = "since";
+    private static final String UNTIL_DATE_ARG = "until";
+
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS = "Failed to parse inputs arguments";
+    private static final String PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE = "Failed due to missing CSV file";
+
+    private static final String DEFAULT_FILE_ARG = ".";
+
+    private File configFile;
+    private File targetFile;
+    private Optional<Date> sinceDate;
+    private Optional<Date> untilDate;
+
+    /**
+     * Parses user-supplied arguments.
+     *
+     * @throws IllegalArgumentException If the given args inputs are malformed or fail to parse or mandatory fields are missing.
+     * @throws ParseException If the given date string fails to parse.
+     */
+    public CliArguments(String[] args) throws IllegalArgumentException, ParseException {
+        sinceDate = Optional.empty();
+        untilDate = Optional.empty();
+        targetFile = new File(DEFAULT_FILE_ARG);
+
+        final HashMap<String, String> argumentMap = new HashMap<String, String>();
+
+        addArgsToArgumentMap(args, argumentMap);
+        checkAllMandatoryArgumentsPresent(argumentMap);
+        setUserInputValuesToArgument(argumentMap);
+    }
+
+    /**
+     * Checks if args begins with a dash character and follows by an input.
+     *
+     * @throws IllegalArgumentException If the given args are malformed,
+     * which means it does not have a dash character or does not follow by an input.
+     */
+    private void addArgsToArgumentMap(String[] args, HashMap<String, String> argumentMap) {
```

Comment:
How about `generateArgumentMap`? So it will return the HashMap object instead of taking it as argument